app/controllers/repositories_controller.rb

```diff
@@ -0,0 +1,9 @@
+class RepositoriesController < ApplicationController
+  def github
```

Comment:
Here you are depending on an external service, is better to use asynchronous request, with this gem 'async_request'.
https://github.com/Wolox/async-requests

Reply:
You will need to do polling of the request in the javascript.